ruby_02-web_applications_with_ruby/outlines/route_helpers.md

```diff
@@ -9,66 +9,87 @@ tags: routes, helpers
+  * Use a link helper to make writing html links easier.
   * Find a routes prefix and use that prefix to build a helper.
 
-### What is CRUD again?
+### Review
+
+What is CRUD again?
 
   * Create
   * Read
   * Update
   * Delete
 
-### What are the specific CRUD routes I get:
+### Warmup
 
-  * If I am modeling the entity `:song`?
-  * If I am modeling the entity `:phone`?
-  * If I am modeling the entity `:product`?
-  * If I am modeling the entity `:sticker`?
+In pairs, one of you write the CRUD routes for `song`, one for `phone`. If you're in a group of three, have the third person write the routes for `sticker`.
 
-##### Do you notice any repetition or patterns here?
+With your partner, do you notice any repetition or patterns in the routes that you wrote?
 
-### Let's go practice hands on
+Large group share.
+
+### Practice
 
 #### Routes
 
-  Clone down [this repo](https://github.com/Carmer/helpers_playground).
+Clone down [this repo](https://github.com/Carmer/helpers_playground).
 
   `git clone https://github.com/Carmer/helpers_playground`
 
-  Now go to the `routes.rb` file.
+Now go to the `routes.rb` file.
+
+Let's create a `resources` line for each of the entities below and then run `rake routes` from the terminal:
+
+  * Song
+  * Phone
+  * Sticker
 
-  Let's create a `resources` for each of the entities below and look at the `rake routes`:
-  
-    * Song
-    * Phone
-    * Product
-    * Sticker
+`rake routes` gives you output in tabular format.
 
-What is a prefix?
-What is a URI pattern?
-What 5 things to we get from rake routes?
+* What are the headers of the table that it gives you?
+* With your partner, take a look at the entries in the table that `rake routes` gives you and fill out the table below in your notebook or on your computer.
+
+|`rake routes` Table Heading|Exampe Entry|Potential Definition|
+|---------------------------|------------|--------------------|
+|                           |            |                    |
+
+Fill in "Potential Definition" with your understanding of what what the column represents/how it can be used. If you're unsure of a definition, enter your best guess.
+
+Large group share.
 
 #### Route Helpers
 
-  * Start your server with `rails s`.
-  * Navigate to localhost:3000/artists. You should see each of the artist name as a link
-  * In your text editor, open up `app/views/artists/index.html.erb`. Right now, the code to make each individual article link looks like this:
+* Start your server with `rails s`.
+* Navigate to localhost:3000/artists. You should see each of the artist name as a link
+* In your text editor, open up `app/views/artists/index.html.erb`. Right now, the code to make each individual article link looks like this:
+
+```html
+  <a href="/artists/<%= artist.id %>"><%= artist.name %></a>
+```
+
+Let's replace the code above with the following:
 
-  ```html
-    <a href="/artists/<%= artist.id %>"><%= artist.name %></a>
-  ```
+```erb
```

Comment:
I think the intent before was to get them to generate the idea of using link_to on their own, instead of being given the solution. Let's see what @case-eee has to say as I think she made that change.